src/solver/optim-model-filler/ComponentFiller.cpp

```diff
@@ -163,10 +163,25 @@ void ComponentFiller::addVariables(Optimisation::LinearProblemApi::ILinearProble
                                                                data);
 
     Expressions::Visitors::EvalVisitor evaluator(evaluationContext, ctx);
+    auto valueOrDefault = [&evaluator](const auto& node, double defaultValue)
+    {
+        if (node.Empty())
+        {
+            return Expressions::Visitors::EvaluationResult(defaultValue);
+        }
+        return evaluator.dispatch(node.RootNode());
+    };
     for (const auto& variable: component_.getModel()->Variables() | std::views::values)
     {
-        const auto& lb = evaluator.dispatch(variable.LowerBound().RootNode());
-        const auto& ub = evaluator.dispatch(variable.UpperBound().RootNode());
+        namespace SM = ModelerStudy::SystemModel;
+        const auto& lb = valueOrDefault(variable.LowerBound(),
+                                        variable.Type() == SM::ValueType::BOOL
+                                          ? 0
+                                          : -std::numeric_limits<double>::infinity());
```

Comment:
Shouldn't we define our own `infinity` ? As each solver may interpret it differently under the hood : for instance I think Xpress will put `1e20` whereas COIN something around `1e300`. This will result in different problems depending on the solvers and possibly behaviors that are solver-dependent

Reply:
FYI Google does not recommend / maintain MPSolver::solver_infinty() : https://github.com/google/or-tools/pull/4313

Reply:
We're already using infinity in a lot of other place though